htdocs/emailcollector/class/emailcollector.class.php

```diff
@@ -2063,7 +2063,7 @@ public function doCollectOneCollector($mode = 0)
 						} elseif (preg_match('/<(.*@.*)>/', $reference, $reg)) {
 							// This is an external reference, we check if we have it in our database
 							if (!is_object($objectemail)) {
-								$sql = "SELECT rowid FROM ".MAIN_DB_PREFIX."ticket where email_msgid = '".$this->db->escape($reg[1])."'";
+								$sql = "SELECT rowid FROM ".MAIN_DB_PREFIX."ticket where email_msgid = '".$this->db->escape($reg[1])."' OR origin_references like '%".$this->db->escape($reg[1])."%'";
```

Comment:
You must also include this->db->escapeforlike into the escape().
Escapeforlike is escaping the % and _ char 
Escape is escaping the '